File: app/src/main/java/com/kozhevin/rootchecks/util/ChecksHelper.java
package com.kozhevin.rootchecks.util;

import android.support.annotation.StringRes;

import com.kozhevin.rootchecks.R;
import com.kozhevin.rootchecks.annotation.CheckingMethodType;

import static com.kozhevin.rootchecks.constant.GeneralConst.CH_TYPE_BUSYBOX_BINARY;
import static com.kozhevin.rootchecks.constant.GeneralConst.CH_TYPE_DANGEROUS_PROPS;
import static com.kozhevin.rootchecks.constant.GeneralConst.CH_TYPE_DEV_KEYS;
import static com.kozhevin.rootchecks.constant.GeneralConst.CH_TYPE_HOOKS;
import static com.kozhevin.rootchecks.constant.GeneralConst.CH_TYPE_NON_RELEASE_KEYS;
import static com.kozhevin.rootchecks.constant.GeneralConst.CH_TYPE_PERMISSIVE_SELINUX;
import static com.kozhevin.rootchecks.constant.GeneralConst.CH_TYPE_RESETPROP;
import static com.kozhevin.rootchecks.constant.GeneralConst.CH_TYPE_SUPER_USER_APK;
import static com.kozhevin.rootchecks.constant.GeneralConst.CH_TYPE_SU_BINARY;
import static com.kozhevin.rootchecks.constant.GeneralConst.CH_TYPE_SU_EXISTS;
import static com.kozhevin.rootchecks.constant.GeneralConst.CH_TYPE_TEST_KEYS;
import static com.kozhevin.rootchecks.constant.GeneralConst.CH_TYPE_UNKNOWN;
import static com.kozhevin.rootchecks.constant.GeneralConst.CH_TYPE_WRONG_PATH_PERMITION;
import static com.kozhevin.rootchecks.constant.GeneralConst.CH_TYPE_XPOSED;

/**
 * The MIT License (MIT)
 * <p>
 * Copyright (c) 2017  Dmitrii Kozhevin <kozhevin.dima@gmail.com>
 * <p>
 * Permission is hereby granted, free of charge, to any person obtaining a copy of this software
 * and associated documentation files (the “Software”), to deal in the Software without
 * restriction, including without limitation the rights to use, copy, modify, merge, publish,
 * distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the
 * Software is furnished to do so, subject to the following conditions:
 * <p>
 * The above copyright notice and this permission notice shall be included in all copies or
 * substantial portions of the Software.
 * <p>
 * THE SOFTWARE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
 * BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
 * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
 * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
 * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

public class ChecksHelper {
    private ChecksHelper() {
    }

    @StringRes
    public static int getCheckStringId(@CheckingMethodType int typeCheck) {
        @StringRes int result = 0;
        switch (typeCheck) {
            case CH_TYPE_TEST_KEYS:
                result = R.string.checks_desc_1;
                break;

            case CH_TYPE_DEV_KEYS:
                result = R.string.checks_desc_2;
                break;

            case CH_TYPE_NON_RELEASE_KEYS:
                result = R.string.checks_desc_3;
                break;

            case CH_TYPE_DANGEROUS_PROPS:
                result = R.string.checks_desc_4;
                break;

            case CH_TYPE_PERMISSIVE_SELINUX:
                result = R.string.checks_desc_5;
                break;

            case CH_TYPE_SU_EXISTS:
                result = R.string.checks_desc_6;
                break;

            case CH_TYPE_SUPER_USER_APK:
                result = R.string.checks_desc_7;
                break;

            case CH_TYPE_SU_BINARY:
                result = R.string.checks_desc_8;
                break;

            case CH_TYPE_BUSYBOX_BINARY:
                result = R.string.checks_desc_9;
                break;

            case CH_TYPE_XPOSED:
                result = R.string.checks_desc_10;
                break;

            case CH_TYPE_RESETPROP:
                result = R.string.checks_desc_11;
                break;

            case CH_TYPE_WRONG_PATH_PERMITION:
                result = R.string.checks_desc_12;
                break;

            case CH_TYPE_HOOKS:
                result = R.string.checks_desc_13;
                break;

            case CH_TYPE_UNKNOWN:
            default:
                result = R.string.empty;
        }
        return result;
    }
}

File: appKotlin/src/main/java/com/kozhevin/rootchecks/util/ChecksHelper.kt
package com.kozhevin.rootchecks.util

import android.content.Context
import android.graphics.Bitmap
import android.graphics.BitmapFactory
import androidx.annotation.StringRes

import com.kozhevin.rootchecks.R

import com.kozhevin.rootchecks.constant.GeneralConst.CH_TYPE_BUSYBOX_BINARY
import com.kozhevin.rootchecks.constant.GeneralConst.CH_TYPE_DANGEROUS_PROPS
import com.kozhevin.rootchecks.constant.GeneralConst.CH_TYPE_DEV_KEYS
import com.kozhevin.rootchecks.constant.GeneralConst.CH_TYPE_HOOKS
import com.kozhevin.rootchecks.constant.GeneralConst.CH_TYPE_NON_RELEASE_KEYS
import com.kozhevin.rootchecks.constant.GeneralConst.CH_TYPE_PERMISSIVE_SELINUX
import com.kozhevin.rootchecks.constant.GeneralConst.CH_TYPE_RESETPROP
import com.kozhevin.rootchecks.constant.GeneralConst.CH_TYPE_SUPER_USER_APK
import com.kozhevin.rootchecks.constant.GeneralConst.CH_TYPE_SU_BINARY
import com.kozhevin.rootchecks.constant.GeneralConst.CH_TYPE_SU_EXISTS
import com.kozhevin.rootchecks.constant.GeneralConst.CH_TYPE_TEST_KEYS
import com.kozhevin.rootchecks.constant.GeneralConst.CH_TYPE_UNKNOWN
import com.kozhevin.rootchecks.constant.GeneralConst.CH_TYPE_WRONG_PATH_PERMITION
import com.kozhevin.rootchecks.constant.GeneralConst.CH_TYPE_XPOSED

object ChecksHelper {

    private var mNonCheck: Bitmap? = null
    private var mFound: Bitmap? = null
    private var mOk: Bitmap? = null

    @StringRes
    fun getCheckStringId(typeCheck: Int): Int = when (typeCheck) {
        CH_TYPE_TEST_KEYS -> R.string.checks_desc_1
        CH_TYPE_DEV_KEYS -> R.string.checks_desc_2
        CH_TYPE_NON_RELEASE_KEYS -> R.string.checks_desc_3
        CH_TYPE_DANGEROUS_PROPS -> R.string.checks_desc_4
        CH_TYPE_PERMISSIVE_SELINUX -> R.string.checks_desc_5
        CH_TYPE_SU_EXISTS -> R.string.checks_desc_6
        CH_TYPE_SUPER_USER_APK -> R.string.checks_desc_7
        CH_TYPE_SU_BINARY -> R.string.checks_desc_8
        CH_TYPE_BUSYBOX_BINARY -> R.string.checks_desc_9
        CH_TYPE_XPOSED -> R.string.checks_desc_10
        CH_TYPE_RESETPROP -> R.string.checks_desc_11
        CH_TYPE_WRONG_PATH_PERMITION -> R.string.checks_desc_12
        CH_TYPE_HOOKS -> R.string.checks_desc_13
        CH_TYPE_UNKNOWN -> R.string.empty
        else -> R.string.empty
    }

    fun getNonCheck(pContext: Context): Bitmap? {
        if (mNonCheck == null || mNonCheck!!.isRecycled) {
            mNonCheck = BitmapFactory.decodeResource(pContext.resources, R.drawable.ic_verified_grey)
        }
        return mNonCheck
    }

    fun getFound(pContext: Context): Bitmap? {
        if (mFound == null || mFound!!.isRecycled) {
            mFound = BitmapFactory.decodeResource(pContext.resources, R.drawable.ic_verified_red)
        }
        return mFound
    }

    fun getOk(pContext: Context): Bitmap? {
        if (mOk == null || mOk!!.isRecycled) {
            mOk = BitmapFactory.decodeResource(pContext.resources, R.drawable.ic_verified_green)
        }
        return mOk
    }
}
